src/ast/ast.ts

```diff
@@ -0,0 +1,435 @@
+import * as vscode from "vscode";
+import { CxWrapper } from "@checkmarxdev/ast-cli-javascript-wrapper";
+import CxScaRealtime from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/scaRealtime/CxScaRealTime";
+import CxScan from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/scan/CxScan";
+import CxProject from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/project/CxProject";
+import CxCodeBashing from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/codebashing/CxCodeBashing";
+import { CxConfig } from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/wrapper/CxConfig";
+import CxLearnMoreDescriptions from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/learnmore/CxLearnMoreDescriptions";
+import {
+  constants
+} from "../utils/common/constants";
+import { getFilePath } from "../utils/utils";
+import { SastNode } from "../models/sastNode";
+import AstError from "../exceptions/AstError";
+import { Logs } from "../models/logs";
+import { CxCommandOutput } from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/wrapper/CxCommandOutput";
+import { ChildProcessWithoutNullStreams } from "child_process";
+import { CxParamType } from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/wrapper/CxParamType";
+
+export async function scaScanCreate(
+  sourcePath: string
+): Promise<CxScaRealtime[] | undefined> {
+  const cx = new CxWrapper(getBaseAstConfiguration());
+  let jsonResults = [];
+  const scan = await cx.runScaRealtimeScan(sourcePath);
+  if (scan.payload && scan.payload.length > 0 && scan.exitCode === 0) {
+    if (scan.payload[0].results) {
+      jsonResults = scan.payload[0].results;
+    }
+  } else {
+    throw new Error(scan.status);
+  }
+
+  return jsonResults;
+}
+
+export async function scanCreate(
+  projectName: string,
+  branchName: string,
+  sourcePath: string
+) {
+  const config = getAstConfiguration();
```

Comment:
Actually we don't because in case there is no API Key populated then the panel switches to the settings panel. So we will never make a call to the platform where the API Key is undefined